### PR TITLE
feat: add improved example project

### PR DESCRIPTION
**Pull request by jeroen-meijer**
## Description

This PR adds an example project that showcases how to test a pair of example widgets.

The `example.md` file is preserved (with some minor change). This way, users on pub.dev will still see a single example file, but may also consult the full example project if they wish.

image

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [x] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+# x.x.x
+
+- feat: add improved example project
```
Comment (marcossevilla) [inline CHANGELOG.md:3]: can we remove this changelog part? we can address that on #53

**Review by marcossevilla — CHANGES_REQUESTED** [review]
lgtm, just a comment regarding changelog

**Comment by Kirpal** [comment]
This is all good, but it would be nice to see some more advanced use cases being tested, such as tapping, injecting app theme/localizations, loading images, etc. What do you think? I don't mind adding that if you'd like me to.

**Comment by alestiago** [comment]
@Kirpal maybe we can have more than one example, similar to how it's being done in the bloc library. 

Thus, if one is not interested in learning about advanced use cases and only wants to know how to get started they can select a simpler example.

**Comment by jeroen-meijer** [comment]
Agreed with @Kirpal and @alestiago on this one.

I can rename this "simple_example" which allows us the opportunity to add more complex ones in the future. How does that sound?

File: CHANGELOG.md (hunk 1 above)
Comment (jeroen-meijer) [reply CHANGELOG.md:3]: Good catch. Will do!

**Comment by Kirpal** [comment]
That sounds good to me! @jeroen-meijer @alestiago

**Comment by jeroen-meijer** [comment]
Will get to this tomorrow! :)

**Comment by marcossevilla** [comment]
@jeroen-meijer are we adding this simple example and then the complex one in a separate PR? just to see if we can approve/merge this one

**Comment by jeroen-meijer** [comment]
I guess we can do the rename later.

Sorry I haven't gotten to this yet, got swamped with other stuff :/
I say we merge this and do the more complex example and the rename later @marcossevilla 👍🏻

**Comment by marcossevilla** [comment]
> I guess we can do the rename later.
> 
> Sorry I haven't gotten to this yet, got swamped with other stuff :/ I say we merge this and do the more complex example and the rename later @marcossevilla 👍🏻

no worries! lgtm 🚀

**Review by marcossevilla — APPROVED** [review]

**Pull request merged**